metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_queries.py

```diff
@@ -175,10 +175,24 @@ def local_temp_path(self) -> pathlib.Path:
         return path
 
     def is_temp_table(self, name: str) -> bool:
-        return any(
+        if any(
             re.match(pattern, name, flags=re.IGNORECASE)
             for pattern in self.config.temporary_tables_pattern
-        )
+        ):
+            return True
+
+        # This is also a temp table if
+        #   1. this name would be allowed by the dataset patterns, and
+        #   2. we have a list of discovered tables, and
+        #   3. it's not in the discovered tables list
```

Comment:
Isn't deleted tables as well fall into this category?
It might be ok to handle those as temp tables.

Reply:
yeah that's fine - generally tables get deleted because they're no longer used, and so bridging over them to generate lineage still makes sense